### PR TITLE
minor code refactor and cleanup in bt_peer_connection

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: src/bt_peer_connection.cpp [hunk 1]
```diff
@@ -664,7 +660,7 @@ namespace {
 		aux::write_uint32(crypto_field, write_buf);
 		aux::write_uint16(pad_size, write_buf); // len (pad)
 
-		std::generate(write_buf.data(), write_buf.data() + pad_size, random_byte);
+		aux::random_bytes({write_buf.data(), pad_size});
```
Comment (arvidn) [inline src/bt_peer_connection.cpp]: I think this span would be better written as: ``write_buf.first(pad_size)``

**Comment by arvidn** [comment]
lgtm, just one minor nit

File: src/bt_peer_connection.cpp (hunk 1 above)
Comment (aldenml) [inline src/bt_peer_connection.cpp]: OK

Comment (aldenml) [inline src/bt_peer_connection.cpp]: Hi @arvidn, do you mind taking a look at the two lines before this one? I see a couple of `aux::write_uint16`, but then, this overwrite whatever is there, is that right? or am I missing something?

Comment (arvidn) [inline src/bt_peer_connection.cpp]: those writes updates the span to refer to the remaining bytes. Perhaps it's not the most intuitive interface, but I think it's at least an improvement over the older versions that just increments a pointer/iterator. i.e. ``write_buf`` is passed by non-const reference there.

Comment (aldenml) [inline src/bt_peer_connection.cpp]: true, true, now I remember :) "the pointer is incremented under the hood"

**Pull request merged**